### PR TITLE
Add OpenCL cumulative sum

### PR DESCRIPTION
**Pull request by t4c1**
## Summary
Adds prim and rev OpenCL implementations for `cumulative_sum`.

## Tests
Both integer and double overloads of the new function are tested.

## Side Effects
None.

## Release notes
OpenCL: Added OpenCL implementation for `cumulative_sum`.

## Checklist

- [ ] Math issue #(issue number)

- [ ] Copyright holder: Tadej Ciglarič

    The copyright holder is typically you or your assignee, such as a university or company. By submitting this pull request, the copyright holder is agreeing to the license the submitted work under the following licenses:
      - Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
      - Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

- [ ] the basic tests are passing

    - unit tests pass (to run, use: `./runTests.py test/unit`)
    - header checks pass, (`make test-headers`)
    - dependencies checks pass, (`make test-math-dependencies`)
    - docs build, (`make doxygen`)
    - code passes the built in [C++ standards](https://github.com/stan-dev/stan/wiki/Code-Quality) checks (`make cpplint`)

- [ ] the code is written in idiomatic C++ and changes are documented in the doxygen

- [ ] the new changes are tested

### CODE REVIEW

**Comment by rok-cesnovar** [comment]
So the change from std::string to const char* was the clang workaround?

**Comment by t4c1** [comment]
Yes. Although I have no idea why or how it worked.

File: stan/math/opencl/kernels/cumulative_sum.hpp [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+#include <stan/math/opencl/buffer_types.hpp>
+#include <stan/math/opencl/matrix_cl_view.hpp>
+#include <string>
+
+namespace stan {
+namespace math {
+namespace opencl_kernels {
+
+// \cond
+static const char *cumulative_sum1_kernel_code = STRINGIFY(
```
Comment (rok-cesnovar) [inline stan/math/opencl/kernels/cumulative_sum.hpp:15]: Could we give them descriptive names as opposed to sum1, sum2? 

If not, maybe a comment on the top of the file with a one line description would be helpful I guess.

Comment (t4c1) [reply stan/math/opencl/kernels/cumulative_sum.hpp:15]: I am not sure what good names would be. If you have a suggestion, go ahead. They already have comments about what do they do in the doxygen. Would you like to see anything more?

**Review by rok-cesnovar — APPROVED** [review]

**Pull request merged**